hack/test-all-minikube-local-registry.sh

```diff
@@ -7,6 +7,30 @@ export KBLD_E2E_DOCKERHUB_HOSTNAME=$(minikube ip):30777
 # uncomment to disable stress tests
 # export KBLD_E2E_SKIP_STRESS_TESTS=true
 
-kapp deploy -a reg -f test/e2e/assets/minikube-local-registry.yml -y
+# Install secretgen to generate registry certs
+kapp deploy -a sg -f https://github.com/vmware-tanzu/carvel-secretgen-controller/releases/download/v0.8.0/release.yml -y
+# Install local docker2 registry
+kapp deploy -a reg -f <(ytt -f test/e2e/assets/minikube-local-registry.yml -v registry_alt_name=$(minikube ip)) -y
+
+if [ ! -z "$GITHUB_RUN_ID" ]; then
+  # Install registry ca cert on the host machine
+  kubectl get secret registry-ca-cert -ojsonpath='{.data.crt\.pem}' | base64 --decode > registry-ca-cert.crt
```

Comment:
I'm not sure why we are adding ca-certs to the registry? Or how this is tied to "supporting docker buildx builder" Esp since docker buildx is configured to insecure (http).

Reply:
this is coming from a commit that switches to linux, so this setup is for other builders. i believe it was bazel that needs that ca-cert configuration since its executing directly on the machine.

Reply:
Ah ok I see. So without adding this the bazel tests (or maybe some other builder test) will fail due to unable to verify the registry cert.

It also looks like the registry previously wasn't configured with TLS. This change will result in our builder tests running against a more 'realistic' registry configuration out in the wild. That is a nice plus :-)

Reply:
However, on the other hand, I'm running into issues when running tests that push / interact with the registry since the registry ca-cert is not provided to the client.

Specifically running `TestDockerBuildxBuildAndPushSuccessful` locally results in `error: Get "https://192.168.59.100:30777/v2/": x509: certificate signed by unknown authority`

Reply:
Yeah, I'm also seeing this failure locally

Reply:
Now that tests are [run within a container](https://github.com/vmware-tanzu/carvel-kbld/blob/develop/hack/test-all-locally.sh) (populated with all the dependencies required), It *should* be possible to remove this `if` statement, since the 'test container' is based off a debian image. 

this should result in always populating ca-certs regardless of whether tests were run in gh-action or locally on a dev workstation, since *both* use the same test container to run all the tests.